src/python/bot/tasks/fuzz_task.py

```diff
@@ -1254,319 +1258,391 @@ def get_strategy_distribution_from_ndb():
-      process_handler.terminate_stale_application_instances()
-      logs.log_error('Unable to create temporary crash queue.')
-      break
-
-    while thread_index < max_threads and test_number < len(testcase_file_paths):
-      testcase_file_path = testcase_file_paths[test_number]
-      gestures = testcases_metadata[testcase_file_path]['gestures']
-
-      env_copy = environment.copy()
-      thread = process_handler.get_process()(
-          target=testcase_manager.run_testcase_and_return_result_in_queue,
-          args=(temp_queue, thread_index, testcase_file_path, gestures,
-                env_copy, True))
-
-      try:
-        thread.start()
-      except:
+  def do_blackbox_fuzzing(self, fuzzer, fuzzer_directory, job_type):
+    """Run blackbox fuzzing."""
+    # Set the thread timeout values.
+    # TODO(ochang): Remove this hack once engine fuzzing refactor is compelte.
+    fuzz_test_timeout = environment.get_value('FUZZ_TEST_TIMEOUT')
+    if fuzz_test_timeout:
+      test_timeout = set_test_timeout(fuzz_test_timeout,
+                                      self.timeout_multiplier)
+    else:
+      test_timeout = self.test_timeout
+
+    thread_timeout = test_timeout
+
+    # Determine number of testcases to process.
+    testcase_count = environment.get_value('MAX_TESTCASES')
+
+    # For timeout multipler greater than 1, we need to decrease testcase count
+    # to prevent exceeding task lease time.
+    if self.timeout_multiplier > 1:
+      testcase_count /= self.timeout_multiplier
+
+    # Run the fuzzer to generate testcases. If error occurred while trying
+    # to run the fuzzer, bail out.
+    (error_occurred, testcase_file_paths,
+     sync_corpus_directory, fuzzer_metadata) = run_fuzzer(
+         fuzzer, fuzzer_directory, self.testcase_directory, self.data_directory,
+         testcase_count)
+
+    if error_occurred:
+      return None, None, None
+
+    fuzzer_binary_name = fuzzer_metadata.get('fuzzer_binary_name')
+    if fuzzer_binary_name:
+      self.fuzz_target = record_fuzz_target(fuzzer.name, fuzzer_binary_name,
+                                            job_type)
+
+    environment.set_value('FUZZER_NAME', self.fully_qualified_fuzzer_name)
+
+    # Synchronize corpus files with GCS
+    if sync_corpus_directory:
+      self.sync_corpus(sync_corpus_directory)
+      environment.set_value('FUZZ_CORPUS_DIR', sync_corpus_directory)
+
+    # Initialize a list of crashes.
+    crashes = []
+
+    # Helper variables.
+    max_threads = utils.maximum_parallel_processes_allowed()
+    needs_stale_process_cleanup = False
+    test_number = 0
+    testcases_before_stale_process_cleanup = environment.get_value(
+        'TESTCASES_BEFORE_STALE_PROCESS_CLEANUP', 1)
+    thread_delay = environment.get_value('THREAD_DELAY')
+    thread_error_occurred = False
+
+    # TODO: Remove environment variable once fuzzing engine refactor is
+    # complete.  Set multi-armed bandit strategy selection distribution as an
```

Comment:
nit: I think  we generally leave one space after a period.

Reply:
Done.